### PR TITLE
refactor(components/tabs): vertical tabs strict mode

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
[AB#2195597](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2195597)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/652/storybook/)

Component Storybooks:

- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/652/storybooks/tabs-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/652/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/652/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/652/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/652/playground/)

File: libs/components/tabs/src/lib/modules/vertical-tabset/vertical-tab.component.ts [hunk 1]
```diff
@@ -60,28 +60,25 @@ export class SkyVerticalTabComponent implements OnInit, OnDestroy {
    * In the next major version, this property will be automatically set to `tab`.
    */
   @Input()
-  public get ariaRole(): string {
-    if (this.isMobile) {
-      return undefined;
-    }
-    return this._ariaRole || 'tab';
+  public get ariaRole(): string | undefined {
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/tabs/src/lib/modules/vertical-tabset/vertical-tab.component.ts]: If this is always a value, does this need to be `| undefined`?

File: libs/components/tabs/src/lib/modules/vertical-tabset/vertical-tab.component.ts [hunk 2]
```diff
@@ -101,92 +98,105 @@ export class SkyVerticalTabComponent implements OnInit, OnDestroy {
...
+  public showTabRightArrow: boolean | undefined;
 
   /**
    * Specifies an ID for the tab.
    * @deprecated Now that the vertical tabs provide aria labels automatically, this input is no longer necessary.
    */
   @Input()
-  public tabId = `sky-vertical-tab-${++nextId}`;
+  public set tabId(value: string | undefined) {
+    this.#_tabId = value || `sky-vertical-tab-${++nextId}`;
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/tabs/src/lib/modules/vertical-tabset/vertical-tab.component.ts]: Should this setter follow the same pattern as [Filter](https://github.com/blackbaud/skyux/pull/661/files#diff-dcf6c169de3594db5f653870f04d80ff4136fe9485d5d011fcd51810c88d356c)?

File: libs/components/tabs/src/lib/modules/vertical-tabset/vertical-tabset.service.ts [hunk 3]
```diff
@@ -35,35 +35,38 @@ export class SkyVerticalTabsetService {
...
   public tabAdded: Subject<SkyVerticalTabComponent> = new Subject();
 
-  public tabClicked: BehaviorSubject<boolean> = new BehaviorSubject(undefined);
+  public tabClicked: ReplaySubject<boolean> = new ReplaySubject(1);
 
-  private _contentAdded = false;
+  #contentAdded = false;
 
-  private _tabsVisible = false;
+  #_tabsVisible = false;
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/tabs/src/lib/modules/vertical-tabset/vertical-tabset.service.ts:43]: These shouldn't have underscores

**Review by Blackbaud-CoreyArcher — CHANGES_REQUESTED** [review]

**Review by Blackbaud-CoreyArcher — APPROVED** [review]

**Pull request merged**